{{ cookiecutter.package_name }}/{{ cookiecutter.module_name }}/plugin.py

```diff
@@ -14,33 +15,38 @@
 # CONFIGURATION
 ########################################
 
-hooks.Filters.CONFIG_DEFAULTS.add_items(
-    [
+config: dict[str, dict[str, t.Any]] = {
+    "defaults": {
         # Add your new settings that have default values here.
         # Each new setting is a pair: (setting_name, default_value).
-        # Prefix your setting names with '{{ cookiecutter.plugin_name|upper|replace('-', '_') }}_'.
-        ("{{ cookiecutter.plugin_name|upper|replace('-', '_') }}_VERSION", __version__),
-    ]
-)
-
-hooks.Filters.CONFIG_UNIQUE.add_items(
-    [
+        "VERSION": __version__,
+    },
+    "unique": {
         # Add settings that don't have a reasonable default for all users here.
         # For instance: passwords, secret keys, etc.
         # Each new setting is a pair: (setting_name, unique_generated_value).
-        # Prefix your setting names with '{{ cookiecutter.plugin_name|upper|replace('-', '_') }}_'.
         # For example:
-        ### ("{{ cookiecutter.plugin_name|upper|replace('-', '_') }}_SECRET_KEY", "{{ '{{' }} 24|random_string {{ '}}' }}"),
-    ]
-)
-
-hooks.Filters.CONFIG_OVERRIDES.add_items(
-    [
+        ## "SECRET_KEY": "{{ '{{' }} 24|random_string {{ '}}' }}",
+    },
+    "overrides": {
         # Danger zone!
         # Add values to override settings from Tutor core or other plugins here.
         # Each override is a pair: (setting_name, new_value). For example:
-        ### ("PLATFORM_NAME", "My platform"),
-    ]
+        ### "PLATFORM_NAME": "My platform",
+    },
+}
+
+
+hooks.Filters.CONFIG_DEFAULTS.add_items(
+    [(f"{{ cookiecutter.plugin_name|upper|replace('-', '_') }}_{key}", value) for key, value in config.get("defaults", {}).items()]
```

Comment:
You're right, but @kdmccormick and I talked about this in the final `DevEX` meeting and he was okay with it since all of the official tutor plugins now follow this convention.

Reply:
:ok_hand:

Reply:
@CodeWithEmad , now that I consider @regisb 's point, I'm actually going to backtrack on what I said. I think it's good to keep the filters front-and-center. It makes the code easier to read, and it makes it more explicit what the names of the filters are.

Tutor plugin configuration is already a full loop around the [configuration clock](http://mikehadlow.blogspot.com/2012/05/configuration-complexity-clock.html) from Open edX configuration, so in order to keep ourselves from moving any further around the clock, I think we should keep the plugin cookiecutter as simple as possible, even if that means some duplication.

Reply:
Much appreciated 🙏🏻